core/base/src/TROOT.cxx

```diff
@@ -656,7 +656,7 @@ TROOT::TROOT() : TDirectory(),
 TROOT::TROOT(const char *name, const char *title, VoidFuncPtr_t *initfunc)
    : TDirectory(), fLineIsProcessing(0), fVersion(0), fVersionInt(0), fVersionCode(0),
      fVersionDate(0), fVersionTime(0), fBuiltDate(0), fBuiltTime(0),
-     fTimer(0), fApplication(0), fInterpreter(0), fBatch(kTRUE), fEditHistograms(kTRUE),
+     fTimer(0), fApplication(0), fInterpreter(0), fBatch(kTRUE), fWeb(""), fIsWeb(0), fEditHistograms(kTRUE),
```

Comment:
`kFalse`, please. No need to initialize `fWeb` explicitly.

Reply:
@Axel-Naumann 
fIsWeb is integer. 0 - off, 1 - normal mode, 2 - batch mode
But I would implement it differently anyway

Reply:
@linev 
Yes fIsWeb has 3 states that's why I made it a short. 
If you have a better idea let me know.

Reply:
> If you have a better idea let me know.

I would do it as two different boolean flags.
Finally, IsWebBatch() should be replaced with IsBatch() flag.